File: hackerrank/easy/0010-diagonal-difference/python/solution.py
## solution.py

from typing import List

def diagonal_difference(arr: List[List[int]]) -> int:
    n = len(arr)
    left = sum(arr[i][i] for i in range(n))
    right = sum(arr[i][n - 1 - i] for i in range(n))
    return abs(left - right)


# --- demo run ---
if __name__ == "__main__":
    matrix = [
        [11, 2, 4],
        [4, 5, 6],
        [10, 8, -12]
    ]
    print(diagonal_difference(matrix))  # 15

File: hackerrank/easy/0010-diagonal-difference/js/solution.mjs
// solution.mjs

export function diagonalDifference(arr) {
  let left = 0;
  let right = 0;
  for (let i = 0; i < arr.length; i++) {
    left += arr[i][i];
    right += arr[i][arr.length - 1 - i];
  }
  return Math.abs(left - right);
}

// --- demo run ---
const matrix = [
  [11, 2, 4],
  [4, 5, 6],
  [10, 8, -12]
];
console.log(diagonalDifference(matrix)); // 15